manager/logbroker/broker.go

```diff
@@ -0,0 +1,73 @@
+package logbroker
+
+import (
+	"sync"
+
+	"github.com/docker/go-events"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/log"
+	"github.com/docker/swarmkit/manager/state/store"
+	"golang.org/x/net/context"
+)
+
+// LogBroker coordinates log subscriptions to services and tasks. Çlients can
+// publish and subscribe to logs channels.
+//
+// Log subscriptions are pushed to the work nodes by creating log subscsription
+// tasks. As such, the LogBroker also acts as an orchestrator of these tasks.
+type LogBroker struct {
+	store       *store.MemoryStore
+	broadcaster *events.Broadcaster
+	nodes       []*api.Node
+	mu          sync.RWMutex
+}
+
+func NewLogBroker() *LogBroker {
+	return &LogBroker{
+		broadcaster: events.NewBroadcaster(),
```

Comment:
If this broadcaster is not closed it will leak a goroutine.

When you close it, make sure the sinks have been removed first, otherwise trying to close it can deadlock. Other code in swarmkit uses a wrapper in `manager/state/watch` that automates this.